### PR TITLE
fix: table shadow when user set sticky

### LINKED ISSUE #28096: Horizontal scroll shadow in Table not showing on header without fixed right column
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/fixed-header-and-scroll-bar-with-the-page-antd486-forked-i6unf)

### Steps to reproduce
- Open demo at https://ant.design/components/table/#components-table-demo-sticky
- See right shadow covers header as well
- Remove last column in code
- Shadow only covers the contents, not the header

### What is expected?
- Shadow should continue over header

### What is actually happening?
The header is on top of the shadow making it not show

| Environment | Info |
|---|---|
| antd | 4.8.6 |
| React | 16.13.1 |
| System | Mac 10.15 |
| Browser | Chrome 86.0.4240.198 |

### PR DESCRIPTION
**Pull request by mochengdx**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/28096

### 💡 Background and solution
Background: This issues that raised by @EvertEt. It caused by use Table component when users set stricky property also have fixed column. For example :
`import { Table } from 'antd';

const columns = [
  {
    title: 'Full Name',
    width: 100,
    dataIndex: 'name',
    key: 'name',
    fixed: 'left',
  },
  {
    title: 'Age',
    width: 100,
    dataIndex: 'age',
    key: 'age',
    fixed: 'left',
  },
  { title: 'Column 1', dataIndex: 'address', key: '1' },
  { title: 'Column 2', dataIndex: 'address', key: '2' },
  { title: 'Column 3', dataIndex: 'address', key: '3' },
  { title: 'Column 4', dataIndex: 'address', key: '4' },
  { title: 'Column 5', dataIndex: 'address', key: '5' },
  { title: 'Column 6', dataIndex: 'address', key: '6' },
  { title: 'Column 7', dataIndex: 'address', key: '7' },
  { title: 'Column 8', dataIndex: 'address', key: '8' },
];

const data = [
  {
    key: '1',
    name: 'John Brown',
    age: 32,
    address: 'New York Park',
  },
  {
    key: '2',
    name: 'Jim Green',
    age: 40,
    address: 'London Park',
  },
];

ReactDOM.render(<Table sticky columns={columns} dataSource={data} scroll={{ x: 1300 }} />, mountNode);`

The reason that sticky-table header's  z-index will set '@table-sticky-zindex'(3) but  'ant-table-container::after' is 2.

The best solution should set [rc-table](https://github.com/react-component/table/blob/master/src/Table.tsx#L704) , for example:
` <div className={isSticky? `${prefixCls}-container ${prefixCls}-sticky` : `${prefixCls}-container`}>{groupTableNode}</div>`
and then you set ant-table-sticky ,like([code](https://github.com/ant-design/ant-design/blob/master/components/table/style/index.less#L606)) ：
` &-sticky {
    &::after {
         z-index: @table-sticky-zindex;
    }
    &-header {
      position: sticky;
      z-index: @table-sticky-zindex;
    }
  .......
`

But rc-table[https://github.com/react-component/table] is as denpency for antd, not sure I can do this.

### 📝 Changelog
this code that I commit is work for this issue. But will set antd-table-contianer 's z-index will be 3 for forever. 

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix shadow issues when user set table has sticky property but set fixed colum.        |
| 🇨🇳 Chinese |     修复具备 sticky 属性和scroll 属性的 Table 在水平滚动过程中，右侧(或左侧)阴影被头覆盖的情况       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by DaoxingHuang** [comment]
Found some CI tools show errors, checked and found it existed before, how to fixed it. Please advices thanks

File: components/table/style/index.less [hunk 1]
```diff
@@ -597,6 +598,7 @@
 
       &::after {
         box-shadow: inset -10px 0 8px -8px darken(@shadow-color, 5%);
+        z-index: @table-sticky-zindex;
```
Comment (xrkffgg) [inline components/table/style/index.less:601]: z-index 放到  box-shadow 前面

Comment (DaoxingHuang) [reply components/table/style/index.less:601]: @xrkffgg  thanks.

**Comment by DaoxingHuang** [comment]
Hi all , how to fix  argos — 17 differences detected, waiting for your decision . I have no ideas for this.

**Comment by yoyo837** [comment]
> Hi all , how to fix argos — 17 differences detected, waiting for your decision . I have no ideas for this.

If the changes is expected, we need to approve it.

**Comment by yoyo837** [comment]
There seems to be some breaking changes.

image

**Comment by DaoxingHuang** [comment]
> There seems to be some breaking changes.
> 
> image
@yoyo837  Dont understand ,i have added you Dingding,  please approve

**Comment by yoyo837** [comment]
@shaodahong  没改到 Descriptions，那一块的UI test 咋不过呢？

**Comment by shaodahong** [comment]
> @shaodahong 没改到 Descriptions，那一块的UI test 咋不过呢？

可能是 master 基准有更新，rebase 下

**Comment by yoyo837** [comment]
> > @shaodahong 没改到 Descriptions，那一块的UI test 咋不过呢？
> 
> 可能是 master 基准有更新，rebase 下

嗷，对哦，这边我没想到。

**Comment by yoyo837** [comment]
混入了其他修改，rebase后使用--force 来push

**Comment by xrkffgg** [comment]
git push -f

**Comment by afc163** [comment]
Closing for no response, please feel free to open new PR.

**Pull request closed without merging**